pyo3-macros-backend/src/from_pyobject.rs

```diff
@@ -61,6 +61,11 @@ impl<'a> Enum<'a> {
                 if maybe_ret.is_ok() {
                     return maybe_ret
                 }
+                if let Err(inner) = maybe_ret {
+                    let gil = Python::acquire_gil();
+                    let py = gil.python();
```

Comment:
I wonder if you are able to use `obj.py()` here to avoid needing to acquire the GIL?